### PR TITLE
Doc/md/guides/mysql: Add doc about CHECK constraint

### PR DESCRIPTION
**Pull request by priteshtm**

### CODE REVIEW

File: doc/md/guides/mysql/check-constraint.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+  id INT PRIMARY KEY,
+  value INT,
+  CHECK (value > 0)
+);
+
+INSERT INTO example (id, value) VALUES (1, -1);  
+-- This will not trigger an error, even though the CHECK constraint specifies that value must be greater than 0
+```
+
+## Managing `CHECK` constraint is easy with Atlas​
```
Comment (hilakashai) [inline doc/md/guides/mysql/check-constraint.md:183]:
```suggestion
## Managing `CHECK` constraints is easy with Atlas​
```

**Review by hilakashai — COMMENTED** [review]
Amazing work, @priteshtm! 🚀

File: doc/md/guides/mysql/check-constraint.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+INSERT INTO example (id, value) VALUES (1, -1);  
+-- This will trigger an "ERROR 3819 (HY000): Check constraint 'example_chk_1' is violated." error
+```
+
+#### You can also use the `MODIFY COLUMN` statement to add a `CHECK` constraint to a column in MySQL. The syntax is as follows:
+
+```sql title="Syntax"
+ALTER TABLE table_name
+MODIFY COLUMN column_name data_type CHECK (column_name condition);
```
Comment (a8m) [inline doc/md/guides/mysql/check-constraint.md:102]:
```suggestion
MODIFY COLUMN column_name data_type CHECK (expression);
```

File: doc/md/guides/mysql/check-constraint.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+```
+
+The condition for a `value_list` can be any valid MySQL expression. If the condition is not met, the insert or update operation will fail.
+
+## Adding and removing the `CHECK` constraint in MySQL
+
+#### To add a `CHECK` constraint to a table in MySQL, use the `ALTER TABLE` statement:
+
+```sql title="Syntax"
+ADD CONSTRAINT constraint_name CHECK (column_name condition);
```
Comment (a8m) [inline doc/md/guides/mysql/check-constraint.md:57]:
```suggestion
ADD CONSTRAINT constraint_name CHECK (expression);
```

File: doc/md/guides/mysql/check-constraint.md [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+### When do we need `CHECK` constraints?
+
+The following are some examples of when you can use `CHECK` constraints:
+
+- A column must be greater than or equal to a specified value.
+- A date field must be in the format of MM/DD/YYYY.
+- An employee’s joining date must not precede their DOB.
+- The value of an employee’s salary must be a positive integer.
+
+The `CHECK` constraint is commonly used along with the NOT NULL constraint for ensuring that a column contains only valid data and doesn’t contain any NULL values. This is particularly important when defining columns that will be referenced by foreign key constraints.
```
Comment (a8m) [inline doc/md/guides/mysql/check-constraint.md:28]:
```suggestion
The `CHECK` constraint is commonly used along with the NOT NULL constraint for ensuring that a column contains only valid data and doesn’t contain any NULL values. 
```

File: doc/md/guides/mysql/check-constraint.md [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+## Syntax
+
+```sql
+CHECK (column_name value_list)
```
Comment (a8m) [inline doc/md/guides/mysql/check-constraint.md:33]: See the syntax here: https://dev.mysql.com/doc/refman/8.0/en/create-table-check-constraints.html

File: doc/md/guides/mysql/check-constraint.md [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+```
+
+You can use the AND keyword between each `value_list` to make the `CHECK` constraint more restrictive. For example:
+
+```sql
+CHECK (column_name value_list AND value_list)
+```
+
+The condition for a `value_list` can be any valid MySQL expression. If the condition is not met, the insert or update operation will fail.
+
```
Comment (a8m) [inline doc/md/guides/mysql/check-constraint.md:51]: I'm not sure what value_list means, but please see the syntax here: https://dev.mysql.com/doc/refman/8.0/en/create-table-check-constraints.html

File: doc/md/guides/mysql/check-constraint.md [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+INSERT INTO example (id, value) VALUES (1, -1);  
+-- This will trigger an "ERROR 3819 (HY000): Check constraint 'example_chk_1' is violated." error
+```
+
+#### If you want to add a `CHECK` constraint to multiple columns, you can use the following syntax:
+
+```sql title="Syntax"
+ALTER TABLE table_name
+ADD CONSTRAINT constraint_name
+CHECK ((column_name1 condition1) AND (column_name2 condition2) AND (column_name3 condition3));
```
Comment (a8m) [inline doc/md/guides/mysql/check-constraint.md:123]: ditto

**Pull request merged**